### PR TITLE
feat: Added support in lifecycle_policy with transition_to_archive

### LINKED ISSUE #22: Add "transition_to_archive" to EFS lifecycle_policy
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality:
  https://github.com/hashicorp/terraform-provider-aws/blob/v5.32.0/CHANGELOG.md

## Describe the solution you'd like.
I would like to add and change "transition_to_archive" value of EFS lifecycle_policy, in addition to "transition_to_ia"  and "transition_to_primary_storage_class"

### PR DESCRIPTION
**Pull request by glavk**
## Description

Add `transition_to_archive` to `lifecycle_policy`.

## Motivation and Context

Fixes #22 

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ module "efs" {
   provisioned_throughput_in_mibps = 256
 
   lifecycle_policy = {
-    transition_to_ia = "AFTER_30_DAYS"
+    transition_to_ia      = "AFTER_30_DAYS"
```
Comment (antonbabenko) [inline README.md:26]: Please revert this line

File: examples/complete/main.tf [hunk 2]
```diff
@@ -106,6 +106,89 @@ module "efs" {
   tags = local.tags
 }
 
+module "efs_archive" {
```
Comment (antonbabenko) [inline examples/complete/main.tf:109]: There is no need to have this new module call in the example. If you want, update the main example.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Also, please update `versions.tf` to require [AWS provider version 5.32](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5320-january-11-2024) or newer.

File: versions.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.42"
+      version = ">= 5.32"
```
Comment (antonbabenko) [inline versions.tf:7]: Please make the same change in other places.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf [hunk 4]
```diff
@@ -18,6 +18,7 @@ resource "aws_efs_file_system" "this" {
 
     content {
       transition_to_ia                    = try(lifecycle_policy.value.transition_to_ia, null)
+      transition_to_archive               = try(lifecycle_policy.value.transition_to_archive, null)
```
Comment (bryantbiggs) [inline main.tf:21]: hmm, why are we not seeing this same change being picked up in the wrappers?

Comment (antonbabenko) [reply main.tf:21]: Because `transition_to_archive` is just a key inside of `var.lifecycle_policy`

Comment (bryantbiggs) [reply main.tf:21]: ah, good point

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 1.6.0 :tada:

**Pull request merged**